includes/discovery/discovery-protocols.inc.php

```diff
@@ -186,7 +190,24 @@
                         }
 
                         if ($skip_discovery === false) {
-                            $remote_device_id = discover_new_device($lldp['lldpRemSysName'], $device, 'LLDP', $interface);
+                            if ($config['discovery_by_ip'] == true) {
+                                echo "Finding IP address for MAC ${lldp['lldpRemChassisId']}\n";
+                                foreach (array_keys($ptopo_array) as $ptopo_key) {
+                                    if (strcmp($ptopo_array[$ptopo_key]['ptopoConnRemoteChassis'], $lldp['lldpRemChassisId']) == 0) {
+                                        $ip_arr = explode(" ", $ptopo_array[$ptopo_key]['ptopoConnAgentNetAddr']);
+                                        
+                                        $a = hexdec($ip_arr[0]);
+                                        $b = hexdec($ip_arr[1]);
+                                        $c = hexdec($ip_arr[2]);
+                                        $d = hexdec($ip_arr[3]);
+                                        
+                                        $discover_hostname = "$a.$b.$c.$d";
+                                    }
+                                 }
+                            } else {
+                                $discover_hostname = $lldp['lldpRemSysName'];
```

Comment:
Is lldpRemSysName always a hostname? If it's not then we should try that along with the new code.